assets/js/app/Data.js

```diff
@@ -9,7 +9,13 @@ LocusZoom.DataSources = function() {
     this.sources = {};
 };
 
+LocusZoom.KnownDataSources = [];
```

Comment:
I agree that data sources should probably become a standard singleton like the rest. I can move that (though it will introduce breaking changes if we change the add/set method names. I can add an extra method with a depreication warning.).

But that still wound't eliminate the KnownDataSources collection. This is the collection of all defined data sources (not necessarily the ones you are using). So we include in the library an object to represent our Recombination Rate API endpoint. That exists whether or not you use it or not. (If you do want to use it, you add it to your DataSources collection). Are you handling this distinction differently with the data layers?

With this current PR, one should never have to use the KnownDataSources collection directly, but can if they need to (if they don't want to use the new extend() method to create custom data sources but still want all the other advantages of deserialization).